### PR TITLE
Update DG, UG and PPP

### PR DESCRIPTION
**Pull request by angkl0**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -249,20 +249,81 @@ Examples of command use:
...
 
 #### Design considerations
 
-* One design consideration involoved marking/unmarking multiple tasks by adding a space before inputting the index of another task.
+* One design consideration involved marking/unmarking multiple tasks by adding a space before inputting the index of another task.
 However, users could forget to input spaces when inputting indexes of multiple tasks, leading to unintended tasks being
-marked/unmarked. This was considered to be an acceptable trade-off as users would be completing tasks one at a time
-most of the time, so a mass mark/unmark feature is a nice-to-have one.
+marked/unmarked without the users even realising it. Limiting the number of tasks that can be marked/unmarked to 1 is thus 
+considered to be an acceptable trade-off.
```
Comment (parnikkapore) [inline docs/DeveloperGuide.md:285]: Looking great! Nice to see that you are revising not-so-great old text as well.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -249,20 +249,81 @@ Examples of command use:
...
+
+CodeConnect allows users to list all of their tasks with `list` and `list time` commands.
+* `list` command lists all the tasks in the order of most recently added task on top.
+* `list time` command lists all the tasks in the order of the task with the earliest deadline on top.
+
+#### Implementation flow
+
+Listing of tasks is facilitated by `ModelManager`. It contains `filteredTaskList` of type `FilteredList`. 
+It also contains `sortedTaskList` of type `SortedList` that wraps `filteredTaskList` in it. `SortedList` contains the
+method SortedList#setComparator(Comparator<? super E> comparator). The above method is called in
```
Comment (parnikkapore) [inline docs/DeveloperGuide.md:299]: Not sure if this line will be rendered properly

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -249,20 +249,81 @@ Examples of command use:
...
+CodeConnect allows users to list all of their tasks with `list` and `list time` commands.
+* `list` command lists all the tasks in the order of most recently added task on top.
+* `list time` command lists all the tasks in the order of the task with the earliest deadline on top.
+
+#### Implementation flow
+
+Listing of tasks is facilitated by `ModelManager`. It contains `filteredTaskList` of type `FilteredList`. 
+It also contains `sortedTaskList` of type `SortedList` that wraps `filteredTaskList` in it. `SortedList` contains the
+method SortedList#setComparator(Comparator<? super E> comparator). The above method is called in 
+`ModelManager#updateSortedTaskList(Comparator<Task> comparator)` that allows for the sorting of `sortedTaskList` with a comparator.
```
Comment (parnikkapore) [inline docs/DeveloperGuide.md:300]: Perhaps mention somewhere that the `sortedTaskList` is what is finally displayed in the UI?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -217,17 +217,27 @@ Example:
...
 
-Searches for tasks by their name or module. The search is done by **matching keywords** in a **case-insensitive** manner.
+Searches for tasks by their name or module.
+
+<div markdown="block" class="alert alert-primary">
+
+:bulb: Note
+* The search is case-insensitive. e.g. `find n/tutorial` will match `Tutorial`
+* The order of the keywords does not matter. e.g. `find n/Set Problem` will match `Problem Set`
+* Partial and full words will be matched. e.g. `find m/2030S` and `find m/CS2030S` will match tasks whole module is `CS2030S`
```
Comment (parnikkapore) [inline docs/UserGuide.md:227]:
```suggestion
* Partial and full words will be matched. e.g. `find m/2030S` and `find m/CS2030S` will match tasks whose module is `CS2030S`
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -319,17 +329,28 @@ Examples:
...
+<div markdown="block" class="alert alert-primary">
+
+:bulb: Note on searching by name or module
+* The search is case-insensitive. e.g. `findc n/hans` will match `Hans`
+* The order of the keywords does not matter. e.g. `findc n/Hans Bo` will match `Bo Hans`
+* Partial and full words will be matched. e.g. `findc m/2030S` and `findc m/CS2030S` will match contacts who are taking `CS2030S`
+* Tasks that match at least one keyword will be returned (i.e. OR search). e.g. `findc n/Hans Bo` will match 
+  `Hans Gruber` and `Bo Yang`
+
+</div>
```
Comment (parnikkapore) [inline docs/UserGuide.md:343]: Should this be removed or link to the box from the find command (as the semantics are the same and should be considered the same)?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -319,17 +329,28 @@ Examples:
...
-* `findc n/John` returns `john`, `John Doe`
-* `findc n/jo` returns `john`, `John Doe`
-* `findc n/jo ja` returns `john`, `james`
-* `findc m/CS1231S` returns `Alex Yeoh`, `David Li`
-* `findc m/1231` returns `Alex Yeoh`, `David Li`
+* `findc n/John` returns all contacts with name that matches `John`. E.g. `john`, `John Doe`
+* `findc n/jo` returns all contacts with name that matches `jo`. E.g. `john`, `John Doe`
+* `findc n/jo ja` returns all contacts with name that matches `jo` or `ja`. E.g. `john`, `james`
+* `findc m/CS1231S CS1101S` returns all contacts that are taking at least a module that matches `CS1231S` or `CS1101S`.
+* `findc m/1S` returns all contacts that are taking at least a module that matches `1S`.
```
Comment (parnikkapore) [inline docs/UserGuide.md:353]:
```suggestion
* `findc m/CS1231S CS1101S` returns all contacts that are taking at least one module that matches `CS1231S` or `CS1101S`.
* `findc m/1S` returns all contacts that are taking at least one module that matches `1S`.
```

File: docs/diagrams/MarkTaskActivityDiagram.puml [hunk 7]
```diff
@@ -1,17 +1,25 @@
 @startuml
 start
 :User enters mark [index] command;
-
-if () then ([invalid index])
-    :Display error message;
+:Input is parsed;
+if () then ([index is not an interger])
```
Comment (parnikkapore) [inline docs/diagrams/MarkTaskActivityDiagram.puml:5]: Perhaps combine all the branches into a single diamond with multiple branches?

File: docs/diagrams/ListTaskSequenceDiagram.puml [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+Model --> ListTaskCommand
+deactivate Model
+
+create CommandResult
+ListTaskCommand -> CommandResult
+activate CommandResult
+
+CommandResult --> ListTaskCommand : result
+deactivate CommandResult
```
Comment (parnikkapore) [inline docs/diagrams/ListTaskSequenceDiagram.puml:84]: Perhaps remove the CommandResult from the diagram to reduce its size?

File: src/main/java/seedu/address/logic/commands/UnmarkTaskCommand.java [hunk 9]
```diff
@@ -15,7 +15,7 @@
 import seedu.address.model.task.TaskName;
 
 /**
- * Marks a task as done.
+ * Unmarks a task as not done.
```
Comment (parnikkapore) [inline src/main/java/seedu/address/logic/commands/UnmarkTaskCommand.java:18]: Nice catch!

**Review by parnikkapore — CHANGES_REQUESTED** [review]
Great job improving both the documentation and the code quality!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (parnikkapore) [reply docs/DeveloperGuide.md:299]: It is, luckily. Closing

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (angkl0) [reply docs/DeveloperGuide.md:300]: Agreed. Added

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -249,20 +249,81 @@ Examples of command use:
...
+`CodeConnectParser#parseCommand()`.
+3. It is determined the type of command is `mark` and a new `MarkTaskCommandParser` is created and `MarkTaskCommandParser#parse`
+is called, parsing and validating the index.
+4. If index is valid, a new `MarkTaskCommand` is created and returned to `LogicManager`.
+5. `LogicManager` calls `MarkTaskCommand#execute()`. If there exists an unmarked task at index 1, 
+`MarkTaskCommand#createMarkedTask()` is called, creating a copy of the task to be marked with its `status` set to `true`.
+6. `Model#setTask()` is then called to replace the task to be marked with the marked copy.
+
+The following sequence diagram below shows the execution of the `mark` command.
+
```
[the remark this answers is not in the document]
Comment (angkl0) [reply docs/DeveloperGuide.md:270]: Removed as well for list section

File: docs/UserGuide.md (hunk 4 above)
Comment (angkl0) [reply docs/UserGuide.md:227]: Fixed

File: docs/diagrams/MarkTaskActivityDiagram.puml (hunk 7 above)
Comment (angkl0) [reply docs/diagrams/MarkTaskActivityDiagram.puml:5]: The if-else branches are staggered to reflect that the checks are done one after another.

File: docs/diagrams/ListTaskSequenceDiagram.puml (hunk 8 above)
Comment (angkl0) [reply docs/diagrams/ListTaskSequenceDiagram.puml:84]: I think it is fine. Diagram is not cluttered due to it + maintain consistency with other sequence diagrams

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -295,7 +293,8 @@ CodeConnect allows users to list all of their tasks with `list` and `list time`
 #### Implementation flow
 
 Listing of tasks is facilitated by `ModelManager`. It contains `filteredTaskList` of type `FilteredList`. 
-It also contains `sortedTaskList` of type `SortedList` that wraps `filteredTaskList` in it. `SortedList` contains the
+It also contains `sortedTaskList` of type `SortedList` that wraps `filteredTaskList` in it. As any changes to `filteredTaskList`
+will be reflected in the `sortedTaskList`, the `Ui` displays `sortedTaskList` in the tasklist panel. SortedList` contains the
```
Comment (parnikkapore) [inline docs/DeveloperGuide.md:297]:
```suggestion
will be reflected in the `sortedTaskList`, the `Ui` displays `sortedTaskList` in the tasklist panel. `SortedList` contains the
```

File: docs/diagrams/MarkTaskActivityDiagram.puml [hunk 12]
```diff
@@ -2,7 +2,7 @@
 start
 :User enters mark [index] command;
 :Input is parsed;
-if () then ([index is not an interger])
+if () then ([index is not an integer])
```
Comment (parnikkapore) [inline docs/diagrams/MarkTaskActivityDiagram.puml:5]: Good catch!

**Review by parnikkapore — COMMENTED** [review]
Looking good! I'll fix that typo during the merge.

**Review by parnikkapore — DISMISSED** [review]

**Review by parnikkapore — APPROVED** [review]

**Pull request merged**